types/d3-sankey/index.d.ts

```diff
@@ -349,6 +349,18 @@ export interface SankeyLayout<Data, N extends SankeyExtraProperties, L extends S
      * @param compare Node comparison function.
      */
     nodeSort(compare: (a: SankeyNode<N, L>, b: SankeyNode<N, L>) => number): this;
+
+    /**
+     * Returns the link comparison function which defaults to undefined.
+     */
+     linkSort(): ((a: SankeyLink<N, L>, b: SankeyLink<N, L>) => number) | undefined;
+
+     /**
+      * Set the link comparison function and return this Sankey layout generator.
+      *
+      * @param compare Link comparison function.
+      */
+      linkSort(compare: (a: SankeyLink<N, L>, b: SankeyLink<N, L>) => number | undefined | null): this;
```

Comment:
From the tests, it looks like this is slightly off.  I think parentheses are needed around the function signature.
```linkSort(compare: ((a: SankeyLink<N, L>, b: SankeyLink<N, L>) => number) | undefined | null): this;```
The parameter can be null or undefined, but the function itself should return a number.

I could be wrong here though.

Reply:
Essentially for the comparison function you have 4 possible values

1.  a value less than 0 if the first link should be above the second
2.  a value less more than 0 if the first link should be below the second
3. `undefined` (default) does nothing
4. `null` the order is fixed by the input

```
If sort is specified, sets the link sort method and returns this Sankey generator. If sort is not specified, returns the current link sort method, which defaults to undefined, indicating that vertical order of links within each node will be determined automatically by the layout. If sort is null, the order is fixed by the input. Otherwise, the specified sort function determines the order; the function is passed two links, and must return a value less than 0 if the first link should be above the second, and a value greater than 0 if the second link should be above the first, or 0 if the order is not specified.
```
https://github.com/d3/d3-sankey#sankey_linkSort


So when I don't add this I get a TS error because I want my sorting function `linkSortComparison` to return all 3 types of values

`sankey().linkSort(linkSortComparison)` <= will return an error if I don't provide `number | undefined | null`

`nodeSort` works exactly the same way and I updated the definition for it as well.


Please let me know what you think